build.sbt

```diff
@@ -117,18 +117,54 @@ lazy val root = (project in file("."))
     )
   )
 
+lazy val scalaVersionDependentSettings = Def.settings(
+  libraryDependencies ++= (if (scalaVersion.value.startsWith("2."))
+                             COMPILE(
+                               scalaVersion("org.scala-lang" % "scala-reflect" % _).value
+                             )
+                           else Seq.empty),
+  scalacOptions := (if (scalaVersion.value.startsWith("2."))
+                       scalacOptions.value
+                     else
+                       scalacOptions
+                         .value
+                         .filterNot(
+                           Set(
+                             "-Ywarn-macros:after",
+
+                             // Ensure we publish an artifact linked to the appropriate Java std library.
+                             "-release:17",
+
+                             // Implicit resolution debug flags.
+                             "-Vimplicits",
+                             "-Vtype-diffs",
+                             // Make all warnings verbose.
+                             "-Wconf:any:warning-verbose"
+                           )
+                         ))
+)
+
+lazy val `test-helpers` = (project in file("test-helpers"))
+  .settings(
+    crossScalaVersions := Seq("2.13.11", "3.3.0"),
+    libraryDependencies ++= TEST(
+      "org.scalatest" %% "scalatest" % scalaTestVersion
+    )
+  )
 lazy val core = (project in file("core"))
   .settings(commonSettings)
   .settings(
     name := "neotypes-core",
+    crossScalaVersions := Seq("2.13.11", "3.3.0"),
     Compile / sourceGenerators += Boilerplate.generatorTask.taskValue,
     libraryDependencies ++=
       PROVIDED(
         "org.neo4j.driver" % "neo4j-java-driver" % neo4jDriverVersion
-      ) ++ COMPILE(
-        scalaVersion("org.scala-lang" % "scala-reflect" % _).value
-      )
+      ),
+    Test / scalacOptions += "-Wconf:cat=other-pure-statement&msg=org.scalatest.Assertion:s"
```

Comment:
This is necessary to use this hack https://github.com/neotypes/neotypes/pull/667/files#r1305977657